<launch>
    <arg name="world_model_name" default="hma_room05"/>
    <arg name="rgb_topic" default="/relay/hsrb/head_rgbd_sensor/rgb/image_rect_color/compressed"/>
    <arg name="camera_info_topic" default="/hsrb/head_rgbd_sensor/rgb/camera_info"/>

    <!-- Omni3dの起動 -->
    <node pkg="tam_dynamic_map" type="recog_furniture_node.py" name="recog_furniture_node" output="screen">
        <param name="world_model_path" type="str" value="$(arg world_model_name).yaml"/>
        <param name="rgb_topic" type="str" value="$(arg rgb_topic)"/>
        <param name="camera_info_topic" type="str" value="$(arg camera_info_topic)"/>
    </node>

    <!-- worldmodelへの登録ノード -->
    <node pkg="tam_dynamic_map" type="create_world_model.py" name="create_world_model" output="screen">
        <param name="world_model_path" type="str" value="$(arg world_model_name).yaml"/>
    </node>

    <!-- Worldmodelの可視化ノード -->
    <node pkg="tam_dynamic_map" type="load_world_model.py" name="load_world_model" output="screen">
        <param name="world_model_path" type="str" value="$(arg world_model_name).yaml"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz_create_world_model" args="-d $(find tam_dynamic_map)/io/create_world_model.rviz"/>

</launch>
